core/src/text/character_style.rs

```diff
@@ -0,0 +1,41 @@
+use crate::pixelcolor::PixelColor;
+
+/// Character style.
+///
+/// This trait is used to modify character styles programmatically, for example, to implement
+/// rendering of text with multiple colors. Applications shouldn't use this trait and instead use
+/// the character style types that are provided by the text renderer, like `MonoTextStyle` and
+/// `MonoTextStyleBuilder` for the integrated font support.
+///
+///  # Implementation notes
+///
+/// Text renderers don't need to support all settings in this trait. All calls to unsupported
+/// setters should be ignored by the implementation. The trait provided empty default
+/// implementations for all setters.
+pub trait CharacterStyle {
```

Comment:
I might be missing some context from the previous discussions that answers this but: would it be useful if each of these methods returned `&mut Self` for easier chaining?

Reply:
I guess it wouldn't hurt. But the intended use-case (e-t calling these one at a time) doesn't call for it.

Reply:
I don't think this is a common pattern for setters. And if @bugadani doesn't need it anyway I would prefer to keep it this way.

Reply:
No worries, just checking. I had the builder pattern in mind but that doesn't really apply here.